v2/src/log_context_property_basic_types.c

```diff
@@ -16,6 +16,13 @@
     { \
         int result; \
         /* Codes_SRS_LOG_CONTEXT_PROPERTY_BASIC_TYPES_01_001: [ If property_value is NULL, LOG_CONTEXT_PROPERTY_TYPE_IMPL(int64_t).to_string shall fail and return -1. ]*/ \
+        /* Codes_SRS_LOG_CONTEXT_PROPERTY_BASIC_TYPES_01_011: [ If property_value is NULL, LOG_CONTEXT_PROPERTY_TYPE_IMPL(uint64_t).to_string shall fail and return -1. ]*/ \
+        /* Codes_SRS_LOG_CONTEXT_PROPERTY_BASIC_TYPES_01_021: [ If property_value is NULL, LOG_CONTEXT_PROPERTY_TYPE_IMPL(int32_t).to_string shall fail and return -1. ]*/ \
+        /* Codes_SRS_LOG_CONTEXT_PROPERTY_BASIC_TYPES_01_031: [ If property_value is NULL, LOG_CONTEXT_PROPERTY_TYPE_IMPL(uint32_t).to_string shall fail and return -1. ]*/ \
+        /* Codes_SRS_LOG_CONTEXT_PROPERTY_BASIC_TYPES_01_041: [ If property_value is NULL, LOG_CONTEXT_PROPERTY_TYPE_IMPL(int16_t).to_string shall fail and return -1. ]*/ \
+        /* Codes_SRS_LOG_CONTEXT_PROPERTY_BASIC_TYPES_01_051: [ If property_value is NULL, LOG_CONTEXT_PROPERTY_TYPE_IMPL(uint16_t).to_string shall fail and return -1. ]*/ \
+        /* Codes_SRS_LOG_CONTEXT_PROPERTY_BASIC_TYPES_01_061: [ If property_value is NULL, LOG_CONTEXT_PROPERTY_TYPE_IMPL(int8_t).to_string shall fail and return -1. ]*/ \
+        /* Codes_SRS_LOG_CONTEXT_PROPERTY_BASIC_TYPES_01_071: [ If property_value is NULL, LOG_CONTEXT_PROPERTY_TYPE_IMPL(uint8_t).to_string shall fail and return -1. ]*/ \
         if (property_value == NULL) \
```

Comment:
>property_value [](http://example.com/codeflow?start=12&length=14)

why is buffer=NULL a valid argument to the function?

Reply:
After reading some more code, it looks like this function wants to emulate snprintf totally, including NULL/0 arguments.

Reply:
yep